src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphEntityDAO.scala

```diff
@@ -14,93 +14,66 @@ import org.broadinstitute.dsde.rawls.model.AttributeConversions._
-    val entityVertex = getEntityVertex(db, workspaceNamespace, workspaceName, entity.entityType, entity.name).getOrElse({
+    val entityVertex = getEntityVertex(workspaceContext.workspaceVertex, entity.entityType, entity.name).getOrElse({
       val newVertex = addVertex(db, VertexSchema.Entity)
       // need to explicitly set name and type so other pipelines can find this vertex,
       // notably in the case of handling cycles
       newVertex.setProperty("name", entity.name)
       newVertex.setProperty("entityType", entity.entityType)
-
-      addEdge(workspace, entity.entityType, newVertex)
+      addEdge(workspaceContext.workspaceVertex, entity.entityType, newVertex)
       newVertex
     })
     entityVertex
   }
 
-  def save(workspaceNamespace: String, workspaceName: String, entity: Entity, txn: RawlsTransaction): Entity = txn withGraph { db =>
+  override def save(workspaceContext: WorkspaceContext, entity: Entity, txn: RawlsTransaction): Entity = txn withGraph { db =>
     // check for illegal dot characters
     validateUserDefinedString(entity.entityType) // do we need to check this here if we're already validating all edges?
     validateUserDefinedString(entity.name)
     entity.attributes.keys.foreach(validateUserDefinedString)
-
-    val workspace = getWorkspaceVertex(db, workspaceNamespace, workspaceName)
-      .getOrElse(throw new IllegalArgumentException("Cannot save entity to nonexistent workspace " + workspaceNamespace + "::" + workspaceName))
-    val entityVertex = getOrCreateVertex(workspace, workspaceNamespace, workspaceName, entity, txn)
-    saveToVertex[Entity](entity, entityVertex, db, WorkspaceName(workspaceNamespace, workspaceName))
+    val entityVertex = getOrCreateVertex(workspaceContext, entity, txn)
+    saveToVertex[Entity](entity, entityVertex, db, workspaceContext)
     entity
   }
 
-  def get(workspaceNamespace: String, workspaceName: String, entityType: String, entityName: String, txn: RawlsTransaction): Option[Entity] = txn withGraph { db =>
-    getEntityVertex(db, workspaceNamespace, workspaceName, entityType, entityName).map(loadEntity(_, workspaceNamespace, workspaceName))
+  override def get(workspaceContext: WorkspaceContext, entityType: String, entityName: String, txn: RawlsTransaction): Option[Entity] = txn withGraph { db =>
+    getEntityVertex(workspaceContext.workspaceVertex, entityType, entityName).map(loadEntity(_,workspaceContext))
   }
 
-  def delete(workspaceNamespace: String, workspaceName: String, entityType: String, entityName: String, txn: RawlsTransaction) = txn withGraph { db =>
-    getEntityVertex(db, workspaceNamespace, workspaceName, entityType, entityName) match {
-      case Some(v) => v.remove()
-      case None => Unit
+  override def delete(workspaceContext: WorkspaceContext, entityType: String, entityName: String, txn: RawlsTransaction) = txn withGraph { db =>
+    getEntityVertex(workspaceContext.workspaceVertex, entityType, entityName) match {
```

Comment:
alternately things that take the workspace Vertex could take WorkspaceContext, that way we ensure that Vertex is the workspace vertex, just taking Vertex can end up being any vertex